node/consensus/src/lib.rs

```diff
@@ -263,14 +263,23 @@ impl<N: Network> Consensus<N> {
             if self.ledger.contains_transmission(&TransmissionID::from(&transaction_id))? {
                 bail!("Transaction '{}' exists in the ledger {}", fmt_id(transaction_id), "(skipping)".dimmed());
             }
-            // Check if the queue is full.
-            if std::mem::size_of_val(&self.transactions_queue.lock()) >= MAX_TX_QUEUE_SIZE {
-                // If the queue is full, return early.
+            // Check the size of the transctions queue.
+            let transactions_queue_size = std::mem::size_of_val(&self.transactions_queue.lock());
```

Comment:
```suggestion
            let transactions_queue_size = self.transactions_queue.lock().len() * (std::mem::size_of::<TransactionKey<N>>() + std::mem::size_of::<Transaction<N>>());
```
this would be more accurate (the current value won't change regardless of length), but still not perfect; I'll provide a precise calculation tomorrow

Reply:
Took a stab: 6cac104c0